java/arcs/core/host/SdkHandleManager.kt

```diff
@@ -0,0 +1,432 @@
+package arcs.core.host
+
+import arcs.core.common.Id
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtOperationAtTime
+import arcs.core.data.RawEntity
+import arcs.core.data.Schema
+import arcs.core.storage.Callbacks
+import arcs.core.storage.Handle as StorageHandle
+import arcs.core.storage.StorageKey
+import arcs.core.storage.api.Entity
+import arcs.core.storage.api.EntitySpec
+import arcs.core.storage.api.Handle
+import arcs.core.storage.api.ReadWriteCollection
+import arcs.core.storage.api.ReadWriteSingleton
+import arcs.core.storage.api.ReadableCollection
+import arcs.core.storage.api.ReadableSingleton
+import arcs.core.storage.api.WritableCollection
+import arcs.core.storage.api.WritableSingleton
+import arcs.core.storage.handle.CollectionImpl
+import arcs.core.storage.handle.HandleManager
+import arcs.core.storage.handle.SetData
+import arcs.core.storage.handle.SetHandle
+import arcs.core.storage.handle.SetOp
+import arcs.core.storage.handle.SingletonData
+import arcs.core.storage.handle.SingletonHandle
+import arcs.core.storage.handle.SingletonImpl
+import arcs.core.storage.handle.SingletonOp
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+typealias Sender = (block: suspend () -> Unit) -> Unit
+typealias SingletonSenderCallbackAdapter<E> =
+    SenderCallbackAdapter<SingletonData<RawEntity>, SingletonOp<RawEntity>, RawEntity?, E?>
+typealias CollectionSenderCallbackAdapter<E> =
+    SenderCallbackAdapter<SetData<RawEntity>, SetOp<RawEntity>, Set<RawEntity>, E>
+
+/**
+ * Wraps a [HandleManager] and creates SDK handles based on [HandleMode], such as
```

Comment:
Do users have to use the HandleManager? Can they use handles directly? How do these handles relate to non-SDK handles?

Could imagine elaborating on these in the doc comment here.

Reply:
Yes, you have to use HandleManager. There are private methods we could make public that take Storage handles directly, we could expose them later, but right now, even those users who don't go through Particle API to access handles are expected to use HandleManager.

Also, HandleManager abstracts away Core vs Android, so it provides additional benefit.